backend/api/Controllers/ReportController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Api.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class ReportController : ControllerBase
+{
+    private readonly ILogger<ReportController> _logger;
+    private readonly ReportService _reportService;
+
+    public ReportController(ILogger<ReportController> logger, ReportService reportService)
+    {
+        _logger = logger;
+        _reportService = reportService;
+    }
+
+    /// <summary>
+    /// List all reports on the asset.
+    /// </summary>
+    /// <remarks>
+    /// Responses are paginated.
+    ///
+    /// Example query:
+    ///
+    ///     /reports?
+    ///
+    /// <para> This query gets all reports (paginated) </para>
+    /// </remarks>
+    /// <returns> List of reports </returns>
+    /// <response code="200"> The list of reports was successfully returned </response>
+    /// <response code="400"> The query is invalid </response>
+    [HttpGet]
+    [ProducesResponseType(typeof(IList<Report>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<Report>>> GetReports()
+    {
+        var reports = await _reportService.ReadAll();
+        return Ok(reports);
+    }
+
+    /// <summary>
+    /// Lookup report by specified id.
+    /// </summary>
+    /// <remarks>
+    /// </remarks>
+    /// <returns> The report, if it exists </returns>
+    /// <response code="200"> Request successful and report returned </response>
+    /// <response code="404"> The requested resource was not found </response>
+    [HttpGet]
+    [Route("{id}")]
+    [ProducesResponseType(typeof(Report), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Report>> GetReportById([FromRoute] string id)
+    {
+        var report = await _reportService.Read(id);
+        if (report == null)
+        {
+            return NotFound();
```

Comment:
Consider adding a message similar as for the other controllers

Reply:
Good catch, will fix